types/terser-webpack-plugin/index.d.ts

```diff
@@ -23,32 +20,104 @@ declare namespace TerserPlugin {
         extractedComments: any;
     }
 
+    interface FileData {
+        readonly filename: string;
+        readonly basename: string;
+        readonly query: string;
+        readonly hash: string;
+    }
+
     interface ExtractCommentOptions {
-        condition: boolean | string | RegExp | ExtractCommentFn | object;
-        filename?: string | FormatFn;
-        banner?: boolean | string | FormatFn;
+        condition: string | RegExp | ExtractCommentFn;
+        filename?: string | FilenameFn;
+        banner?: false | string | FormatFn;
```

Comment:
Changing boolean to false introduces a breaking change which is not necessary. Why not stay with boolean here?

Reply:
agree, tbd

Reply:
adressed with 9c4abf7e33